beacon-chain/db/filesystem/data_column.go

```diff
@@ -0,0 +1,966 @@
+		if err != nil {
+			return nil, errors.Wrap(err, "unmarshal SSZ encoded data column sidecar")
+		}
+
+		// Create a RO data column.
+		roDataColumnSidecar, err := blocks.NewRODataColumn(dataColumnSidecar)
+		if err != nil {
+			return nil, errors.Wrap(err, "new read only data column")
+		}
+
+		// Create a verified RO data column.
+		verifiedRODataColumn := blocks.NewVerifiedRODataColumn(roDataColumnSidecar)
+
+		// Append the verified RO data column to the data column sidecars.
+		verifiedRODataColumnSidecars = append(verifiedRODataColumnSidecars, verifiedRODataColumn)
+	}
+
+	dataColumnFetchLatency.Observe(float64(time.Since(startTime).Milliseconds()))
+
+	return verifiedRODataColumnSidecars, nil
+}
+
+// Remove deletes all data column sidecars for a given root.
+func (dcs *DataColumnStorage) Remove(blockRoot [fieldparams.RootLength]byte) error {
+	dcs.pruneMu.RLock()
+	defer dcs.pruneMu.RUnlock()
+
+	fileMu, _ := dcs.fileMutex(blockRoot)
+	fileMu.Lock()
+	defer fileMu.Unlock()
+
+	summary, ok := dcs.cache.get(blockRoot)
+	if !ok {
+		// Nothing found in db. Exit early.
+		return nil
+	}
+
+	// Remove the data column sidecars from the cache.
+	dcs.cache.evict(blockRoot)
+
+	// Remove the data column sidecars file.
+	filePath := filePath(blockRoot, summary.epoch)
+	if err := dcs.fs.Remove(filePath); err != nil {
+		return errors.Wrap(err, "remove")
+	}
+
+	return nil
+}
+
+// Clear deletes all files on the filesystem.
+func (dcs *DataColumnStorage) Clear() error {
+	dcs.pruneMu.Lock()
+	defer dcs.pruneMu.Unlock()
+
+	dirs, err := listDir(dcs.fs, ".")
+	if err != nil {
+		return errors.Wrap(err, "list dir")
+	}
+
+	dcs.cache.clear()
+
+	for _, dir := range dirs {
+		if err := dcs.fs.RemoveAll(dir); err != nil {
+			return errors.Wrap(err, "remove all")
+		}
+	}
+
+	return nil
+}
+
+// prune clean the cache, the filesystem and mutexes.
+func (dcs *DataColumnStorage) prune() {
+	dcs.mu.Lock()
+	defer dcs.mu.Unlock()
+
+	highestStoredEpoch := dcs.cache.HighestEpoch()
+
+	// Check if we need to prune.
+	if highestStoredEpoch < dcs.retentionEpochs {
+		return
+	}
+
+	highestEpochToPrune := highestStoredEpoch - dcs.retentionEpochs
+	highestPeriodToPrune := period(highestEpochToPrune)
+
+	// Prune the cache.
+	prunedCount := dcs.cache.pruneUpTo(highestEpochToPrune)
+
+	if prunedCount == 0 {
+		return
+	}
+
+	dataColumnPrunedCounter.Add(float64(prunedCount))
+
+	// Prune the filesystem.
+	periodFileInfos, err := afero.ReadDir(dcs.fs, ".")
+	if err != nil {
+		log.WithError(err).Error("Error encountered while reading top directory")
+		return
+	}
+
+	for _, periodFileInfo := range periodFileInfos {
+		periodStr := periodFileInfo.Name()
+		period, err := strconv.ParseUint(periodStr, 10, 64)
+		if err != nil {
+			log.WithError(err).Errorf("Error encountered while parsing period %s", periodStr)
+			continue
+		}
+
+		if period > highestPeriodToPrune {
+			// Nothing to prune.
+			continue
+		}
+
+		if period == highestPeriodToPrune {
```

Comment:
Don't you want `<=` here?

Reply:
Done in https://github.com/prysmaticlabs/prysm/pull/15061/commits/c7bd2d11b2c4b326645654415a224565b10257b7.